toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -217,6 +217,65 @@ func SetupLoopbackDevice(diskFilePath string) (devicePath string, err error) {
 	return
 }
 
+// BlockOnDiskIO waits until all outstanding operations against a disk complete.
+func BlockOnDiskIO(diskDevPath string) (err error) {
+
+	logger.Log.Infof("Flushing all IO to disk for %s", diskDevPath)
+	shell.Execute("sync")
+
+	stdout, stderr, err := shell.Execute("lsblk", "--output", "MAJ:MIN", "--noheadings", "--nodeps", diskDevPath)
+	if err != nil {
+		logger.Log.Errorf("Failed to get disk major/minor ID: %s, %s", stderr, err.Error())
+		return
+	}
+	// lsblk includes whitespace both before an after output, trim it off.
+	diskIDs := strings.Split(strings.TrimSpace(stdout), ":")
+	if len(diskIDs) != 2 {
+		return fmt.Errorf("couldn't find disk IDs for %s (%s)", diskDevPath, stdout)
+	}
+	maj := diskIDs[0]
+	min := diskIDs[1]
+
+	outstandingOps := ""
+	logger.Log.Tracef("Searching /proc/diskstats for %s:%s", maj, min)
+	for busy := true; busy; busy = (outstandingOps != "0") {
+		outstandingOps = ""
```

Comment:
Is the reassignment necessary? It seems we either always give it a new value or error out.